README.md

```diff
@@ -0,0 +1,248 @@
+
+NGINX supports static and dynamic modules. Static modules are defined at build-time and compiled into the resulting binaries. [Dynamic modules](https://nginx.org/en/linux_packages.html#dynmodules) (for example, [njs](https://github.com/nginx/njs)) are built and distributed separately. They can be added to, or removed from, an NGINX installation at any time.
+
+> [!IMPORTANT]
+> Official NGINX package distributions are built with all native open source static modules.
+
+## Configurations
+NGINX is highly flexible and configurable. Provisioning the software is achieved via text-based config file(s) organized in functional sections called "Contexts", accepting a vast amount of configuration parameters called "[Directives](https://nginx.org/en/docs/dirindex.html)". See [Configuration File's Structure](https://nginx.org/en/docs/beginners_guide.html#conf_structure) for a comprehensive definition of Directives and Contexts.
+
+> [!NOTE]
+> The set of directives available to your distribution of NGINX is dependent on which [modules](#modules) have been made available to it.
+
+## Runtime
+Rather than running in a single, monolithic process, NGINX is architected to scale beyond OS process stack limitations by operating as a collection of processes. They include:
+- A "master" process that maintains worker processes, as well as, reads and evaluates configuration files.
+- One or more "worker" processes that process data (eg. HTTP requests).
+
+The number of [worker processes](https://nginx.org/en/docs/ngx_core_module.html#worker_processes) is typically set to the number of CPU cores on the system. In most cases, this optimally balances load across available system resources, as NGINX is designed to efficiently distribute work across all available worker processes.
+
+> [!TIP]
+> Processes synchronize data through shared memory. For this reason, many NGINX directives require the allocation of shared memory zones. As an example, when configuring [rate limiting](https://nginx.org/en/docs/http/ngx_http_limit_req_module.html#limit_req), connecting clients must be tracked in a [common memory zone](https://nginx.org/en/docs/http/ngx_http_limit_req_module.html#limit_req_zone) so all worker processes can know how many times a particular client has accessed the server in a span of time.
+
+# Downloading and installing
+Follow these steps to download and install precompiled NGINX binaries. You may also choose to [build the module locally from source code](#building-from-source).
+
+## Stable and Mainline binaries
+NGINX binaries are built and distributed in two versions: stable and mainline. You'll need to [decide which is appropriate for your purposes](https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-open-source/#choosing-between-a-stable-or-a-mainline-version).
```

Comment:
Best-practices dictate that information on downloading, installing, basic "getting started" information is included in the Readme. (ie. not just details on how to work with source code). For this reason, I thought it would be important to cover the different binary distros here.

Reply:
It's a bit out of context since this is a subset of the "Downloading and installing" section, but agree there's value in relating the two. For that reason, I've turned it into a tip. https://github.com/nginx/nginx/pull/120/commits/1b40646563b667105fa664844421d797ddc47f33

Reply:
I don't see how this is relevant to a user who came to see instructions for downloading and installing binaries. Which one is the single stable branch? It would be good to link to it. https://github.com/nginx/nginx/branches/all.

Reply:
As a compromise:

Stable binaries are built from stable branches and only contain critical fixes backported from the mainline version.  Mainline are built from the master branch and contain the latest features and bugfixes.

Would that work?

Reply:
Yep, that works. Updated here https://github.com/nginx/nginx/pull/120/commits/baaf16388fa1654e562c9d54d182f8acf3682057